src/test/java/testapp/adapter/http/UserAccountServiceTest.java

```diff
@@ -0,0 +1,168 @@
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import testapp.core.AccountException;
+import testapp.core.AccountRepository;
+import testapp.core.UserAccount;
+
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+import static testapp.matchers.SitebricksMatchers.returns;
+
+public class UserAccountServiceTest {
+    public JUnitRuleMockery context = new JUnitRuleMockery();
+    private LocalServiceTestHelper helper =
+            new LocalServiceTestHelper(new LocalMemcacheServiceTestConfig(), new LocalDatastoreServiceTestConfig());
+
+    @Before
+    public void setUp() throws Exception {
+        helper.setUp();
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        helper.tearDown();
+    }
+
+    @Test
+    public void userGetBalance() throws Exception {
+        final AccountRepository accountRepository = context.mock(AccountRepository.class);
+
+        UserAccountService accountService = new UserAccountService(accountRepository);
+
+        context.checking(new Expectations() {{
+            oneOf(accountRepository).findUserAccount(1);
+            will(returnValue(new UserAccount()));
+        }});
+
+        Reply<?> response = accountService.getUserBalance(1);
+        Reply<?> balance = Reply.with(0).as(Json.class);
+
+        assertThat(response, returns(balance));
+    }
+
+    @Test
+    public void getBalanceOfNonExistingUser() throws Exception {
+        final AccountRepository accountRepository = context.mock(AccountRepository.class);
+
+        UserAccountService accountService = new UserAccountService(accountRepository);
+
+        context.checking(new Expectations() {{
+            oneOf(accountRepository).findUserAccount(1);
+            will(throwException(new AccountException("")));
+        }});
+
+        Reply<?> response = accountService.getUserBalance(1);
+        Reply<?> userDoesNotExist = Reply.saying().badRequest();
+
+        assertThat(response, returns(userDoesNotExist));
+    }
+
+    @Test
+    public void depositToAccount() throws Exception {
+        final AccountRepository accountRepository = context.mock(AccountRepository.class);
+        final Request request = context.mock(Request.class);
+
+        final UserAccount account = new UserAccount(1);
+        UserAccountService accountService = new UserAccountService(accountRepository);
+
+        context.checking(new Expectations() {{
+            oneOf(accountRepository).findUserAccount(1);
+            will(returnValue(account));
+
+            oneOf(request).param("amount");
+            will(returnValue("20"));
+
+            oneOf(accountRepository).updateUserAccount(account);
+        }});
+
+
+        Reply<?> response = accountService.deposit(1, request);
+        Reply<?> userGotNewBalance = Reply.saying().ok();
+
+        int userBalance = account.getTokenBalance();
+
+        assertThat(response, returns(userGotNewBalance));
```

Comment:
here you can assert that the response is ok(200), no need to make a reply, you can have custom matchers for the response codes if you need them